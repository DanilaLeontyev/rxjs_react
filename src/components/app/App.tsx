import React from 'react';
import './App.css';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import ChatPage from '../chat_page/ChatPage';
import { PersonSwitcher } from '../person_switcher/PersonSwitcher';

const acc1 = {
  name: 'first-person'
}

const acc2 = {
  name: 'second-person'
}

const App: React.FC = () => {
  return (
    <Router>
      <>
        <PersonSwitcher />
        <Switch>
          <Route exact path="/" component={() => <ChatPage person={acc1} />} />
          <Route exact path="/first-person" component={() => <ChatPage person={acc1} />} />
          <Route exact path="/second-person" component={() => <ChatPage person={acc2} />} />
        </Switch>
      </>
    </Router>
  );
}

export default App;
