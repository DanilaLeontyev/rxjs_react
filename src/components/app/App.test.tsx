import React from 'react';
import { render } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import App from './App';

test('renders hello', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText('Hello');
  expect(linkElement).toBeInTheDocument();
});

test('let`s render App', () => {
  const component = TestRenderer.create(
    <App />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});