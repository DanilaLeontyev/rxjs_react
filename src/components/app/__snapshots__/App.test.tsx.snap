// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `null`;

exports[`Link changes the class when hovered 1`] = `
<div>
  Hello
</div>
`;

exports[`let 1`] = `null`;

exports[`let s render App 1`] = `null`;

exports[`let\`s render App 1`] = `null`;

exports[`let\`ss render App 1`] = `null`;

exports[`lets render App 1`] = `null`;
